integrations/ivy/src/app/app.component.spec.ts

```diff
@@ -0,0 +1,11 @@
+import { ɵivyEnabled as ivyEnabled } from '@angular/core';
+
+describe('NGXS with Ivy enabled', () => {
+  it('ivy has to be enabled', () => {
+    // This assertion has to be performed as we have to
+    // be sure that we're running these tests with the Ivy engine
+    expect(ivyEnabled).toBeTruthy();
+  });
+
+  // here we can write next tests for Ivy
```

Comment:
```ts
it('with warn', () => {
  const spy = jest.spyOn(console, 'warn').mockImplementation();
  
  @State({ name: 'todo' })
  class TodoState {}

   // etc

   try {
        // Assert
        const UNDECORATED_STATE_IN_IVY = CONFIG_MESSAGES[VALIDATION_CODE.UNDECORATED_STATE_IN_IVY]('todo');
        expect(spy).toHaveBeenCalledWith(UNDECORATED_STATE_IN_IVY);
      } finally {
        spy.mockRestore();
      }

});
```

```ts
it('without warn', () => {
  const spy = jest.spyOn(console, 'warn').mockImplementation();
  
  @State({ name: 'todo' })
  @Injectable()
  class TodoState {}

   // etc

   try {
        // Assert
        const UNDECORATED_STATE_IN_IVY = CONFIG_MESSAGES[VALIDATION_CODE.UNDECORATED_STATE_IN_IVY]('todo');
        expect(spy).not.toHaveBeenCalledWith(UNDECORATED_STATE_IN_IVY);
      } finally {
        spy.mockRestore();
      }

});
```

Reply:
I cannot add this test right now. I will show you later why it fails. (P.S. it fails because of Jest+Ivy).

Reply:
👎 Jest weirdness!